internal/olm/operator/registry/operator_installer.go

```diff
@@ -226,3 +235,53 @@ func (o OperatorInstaller) getInstalledCSV(ctx context.Context) (*v1alpha1.Clust
 	}
 	return csv, nil
 }
+
+// Approve approves the install plan for a subscription, which will
+// generate a CSV
+func (c IndexImageCatalogCreator) Approve(ctx context.Context, sub *v1alpha1.Subscription) error {
```

Comment:
Going along with my other comments, I think this function should be moved to the `OperatorInstaller` struct and renamed to `approveInstallPlan`.

The way the catalog source gets created is decoupled from he creation of the subscription and install plans, so we don't need separate approval logic.

Reply:
Updated the PR to remove the interface and make `approveInstallPlan` a method on `OperatorInstaller` struct, which will be common across both `run packagemanifests` and `run bundle`

Thanks so much for the feedback @joelanford @estroz, please review new changes